CONTRIBUTING.md

```diff
@@ -237,47 +391,123 @@ changes:
-6. Update any documentation, user and contributor, that is impacted by your changes.
-7. You may merge the pull request in once you have the sign-off from one developer from the [Retrospectives team](retrospectives@microsoft.com), or if you do not have permission to do that, you may request the reviewer to merge it for you.
+#### Framework
 
-## Code
+React Component tests are written using the following packages:
+
+- [Jest Testing Framework](https://jestjs.io/)
+- [Enzyme Testing utility](https://enzymejs.github.io/enzyme/)
+- [Enzyme to JSON](https://github.com/adriantoine/enzyme-to-json)
+
+---
+
+#### Test Coverage
+
+To automatically generate the test coverage report, add the `--coverage` flag
+to the `test` script defined in
+[package.json](RetrospectiveExtension.Frontend/package.json). After the test
+run is completed, coverage statistics will then be reported in the newly
+created `coverage` directory.
+
+---
+
+#### Test Execution
+
+- `npm install` must be executed before running any tests.
+- `npm run test` is the default test execution method defined in the
+[package.json](RetrospectiveExtension.Frontend/package.json) file. This will
+automatically run all of the tests in files suffixed with `.test.tsx` inside of
+the [tests folder](RetrospectiveExtension.Frontend/components/__tests__).
+- `npm run test:watch` will run tests in watch mode, re-running tests every
+time a component change is saved.
+- `jest --env=jsdom --silent -ci --testResultsProcessor=jest-junit {FULL_FILE_PATH}`
+can be used to run tests only in the specified file. Wildcards also work
+instead of a fully qualified path.
+
+---
+
+#### Mocks
+
+- In this project, mocks have been implemented for simulating API calls and
+external module functionality. Reusable mocks should be added to the
+[mock folder](RetrospectiveExtension.Frontend/components/__mocks__).
+- Mocks which are shared by the majority of tests should be initialized in the
+[test setup file](RetrospectiveExtension.Frontend/components/__tests__/setupTests.tsx).
+
+---
+
+#### Snapshots
+
+To ensure proper rendering of components, snapshots tests are being used
+to compare expected component rendering state against its actual state.
+Snapshot tests will fail when changes are made to components that are not
+accounted for through updates to these stored snapshots.
+
+To update snapshots, delete the snapshot for the component you are testing,
+(located in the [snapshots folder](RetrospectiveExtension.Frontend/components/__tests__/__snapshots__))
+and run the test command. On test run completion, new snapshots should be
+created. Please check the newly created snapshot file, to ensure that the
+expected changes are present, and include the snapshot in your pull request.
+
+## Backend Development
+
+### Style Guide
+
+Follow the coding guidelines here - [C# Coding Conventions (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions)..
```

Comment:
nit: two periods

Reply:
Do we have a style guide named for the node/typescript portions? Worth mentioning here?

Reply:
🤔 hmm. I can add a note about ESLint since that does much of the style enforcement for typescript